core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -299,9 +298,9 @@ public Node start() {
         injector.getInstance(MonitorService.class).start();
         injector.getInstance(RestController.class).start();
 
-        assert injector.getInstance(ClusterService.class) instanceof InternalClusterService :
+        assert injector.getInstance(ClusterService.class) instanceof ClusterService :
                 "node cluster service implementation must inherit from InternalClusterService";
```

Comment:
Nit: The text should say `"node cluster service implementation must inherit from ClusterService"`